buildstockbatch/gcp/gcp.py

```diff
@@ -379,21 +559,165 @@ def run_batch(self):
+        :param gcs_prefix: Prefix used for GCS files
         """
-        pass
+        # Local directory where we'll write files
+        sim_dir = pathlib.Path('/var/simdata/openstudio')
+        # Mounted GCS bucket directory
+        parent_dir = pathlib.Path('/mnt/disks/share') / gcs_prefix
+
+        logger.info('Extracting assets TAR file')
+        # Copy assets file to local machine to extract TAR file
+        assets_file_path = sim_dir / 'assets.tar.gz'
+        shutil.copyfile(parent_dir / 'assets.tar.gz', assets_file_path)
+        with tarfile.open(assets_file_path, 'r') as tar_f:
+            tar_f.extractall(sim_dir)
+
+        logger.debug('Reading config')
+        with open(parent_dir / 'config.json') as f:
+            cfg = json.load(f)
+
+        # Extract the job information for this particular task
+        logger.debug('Getting job information')
+        jobs_file_path = parent_dir / 'jobs.tar.gz'
+        with tarfile.open(jobs_file_path, 'r') as tar_f:
+            jobs_d = json.load(tar_f.extractfile(f'jobs/job{task_index:05d}.json'), encoding='utf-8')
+        logger.debug('Number of simulations = {}'.format(len(jobs_d['batch'])))
+
+        logger.debug('Getting weather files')
+        weather_dir = sim_dir / 'weather'
+        os.makedirs(weather_dir, exist_ok=True)
+
+        # Make a lookup of which parameter points to the weather file from options_lookup.tsv
+        with open(sim_dir / 'lib' / 'resources' / 'options_lookup.tsv', 'r', encoding='utf-8') as f:
+            tsv_reader = csv.reader(f, delimiter='\t')
+            next(tsv_reader)  # skip headers
+            param_name = None
+            epws_by_option = {}
+            for row in tsv_reader:
+                row_has_epw = [x.endswith('.epw') for x in row[2:]]
+                if sum(row_has_epw):
+                    if row[0] != param_name and param_name is not None:
+                        raise RuntimeError(
+                            'The epw files are specified in options_lookup.tsv under more than one parameter '
+                            f'type: {param_name}, {row[0]}'
+                        )  # noqa: E501
+                    epw_filename = row[row_has_epw.index(True) + 2].split('=')[1]
+                    param_name = row[0]
+                    option_name = row[1]
+                    epws_by_option[option_name] = epw_filename
+
+        # Look through the buildstock.csv to find the appropriate location and epw
+        epws_to_download = set()
+        building_ids = [x[0] for x in jobs_d['batch']]
+        with open(sim_dir / 'lib' / 'housing_characteristics' / 'buildstock.csv', 'r', encoding='utf-8') as f:
+            csv_reader = csv.DictReader(f)
+            for row in csv_reader:
+                if int(row['Building']) in building_ids:
+                    epws_to_download.add(epws_by_option[row[param_name]])
+
+        # Copy files to local machine and unzip the epws needed for these simulations
```

Comment:
Isn't this just extracting from the mounted dir to the local machine (i.e., not actually _copying_ the file to local before extracting)?